### PR TITLE
Arf/optimise

### PR DESCRIPTION
**Pull request by alexfalkowski**
#37

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -2,6 +2,7 @@ package moocro
 
 // Server defines an inteface for all servers
 type Server interface {
+	Close() error
```
[the remark this answers is not in the document]
Comment (alexfalkowski) [reply server.go]: Nice!

File: amqp_connection.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	publisher, err := connect()
+	if err != nil {
+		consumer.Close()
+		return nil, err
+	}
+
+	return &amqpConnection{consumer: consumer, publisher: publisher}, nil
+}
+
+func (c *amqpConnection) consumnerChannel() (*amqp.Channel, error) {
```
Comment (nerde) [inline amqp_connection.go]: Typo "consumNer" xD

File: amqp_channel.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	channel.NotifyClose(notification)
+
+	newChannel := &amqpChannel{channel: channel, connection: c, logger: logger, notification: notification}
+
+	go newChannel.notify()
+
+	return newChannel, nil
+}
+
+func (c *amqpChannel) notify() {
```
Comment (nerde) [inline amqp_channel.go]: I would call this `listenToNotifications`.

File: amqp_connection.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+func (c *amqpConnection) channel() (*amqp.Channel, error) {
+	ch, err := c.connection.Channel()
+	if err != nil {
+		return nil, err
+	}
+
+	return ch, nil
+}
+
+func (c *amqpConnection) notify() {
```
Comment (nerde) [inline amqp_connection.go]: `listenToNotifications`?

File: amqp_connection.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+	newConnection := &amqpConnection{connection: connection, logger: logger, notification: notification}
+
+	go newConnection.notify()
+
+	return newConnection, nil
+}
+
+func (c *amqpConnection) channel() (*amqp.Channel, error) {
+	ch, err := c.connection.Channel()
```
Comment (nerde) [inline amqp_connection.go]: `return c.connection.Channel()`

**Pull request merged**